File: java-vertx/src/main/java/io/swagger/server/api/model/V3ResultRoute.java
package io.swagger.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.server.api.model.V3RouteServiceStatus;
import io.vertx.core.json.JsonObject;
import io.vertx.codegen.annotations.DataObject;
/**
 * V3ResultRoute
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaVertXServerCodegen", date = "2024-04-14T03:35:56.953023300+10:00[Australia/Sydney]")



public class V3ResultRoute   {
  private String routeName = null;

  private String routeNumber = null;

  private Integer routeType = null;

  private Integer routeId = null;

  private String routeGtfsId = null;

  private V3RouteServiceStatus routeServiceStatus = null;

  public V3ResultRoute routeName(String routeName) {
    this.routeName = routeName;
    return this;
  }

  /**
   * Name of route
   * @return routeName
   **/
    public String getRouteName() {
    return routeName;
  }

  public void setRouteName(String routeName) {
    this.routeName = routeName;
  }

  public V3ResultRoute routeNumber(String routeNumber) {
    this.routeNumber = routeNumber;
    return this;
  }

  /**
   * Route number presented to public (nb. not route_id)
   * @return routeNumber
   **/
    public String getRouteNumber() {
    return routeNumber;
  }

  public void setRouteNumber(String routeNumber) {
    this.routeNumber = routeNumber;
  }

  public V3ResultRoute routeType(Integer routeType) {
    this.routeType = routeType;
    return this;
  }

  /**
   * Transport mode identifier
   * @return routeType
   **/
    public Integer getRouteType() {
    return routeType;
  }

  public void setRouteType(Integer routeType) {
    this.routeType = routeType;
  }

  public V3ResultRoute routeId(Integer routeId) {
    this.routeId = routeId;
    return this;
  }

  /**
   * Route identifier
   * @return routeId
   **/
    public Integer getRouteId() {
    return routeId;
  }

  public void setRouteId(Integer routeId) {
    this.routeId = routeId;
  }

  public V3ResultRoute routeGtfsId(String routeGtfsId) {
    this.routeGtfsId = routeGtfsId;
    return this;
  }

  /**
   * GTFS Identifer of the route
   * @return routeGtfsId
   **/
    public String getRouteGtfsId() {
    return routeGtfsId;
  }

  public void setRouteGtfsId(String routeGtfsId) {
    this.routeGtfsId = routeGtfsId;
  }

  public V3ResultRoute routeServiceStatus(V3RouteServiceStatus routeServiceStatus) {
    this.routeServiceStatus = routeServiceStatus;
    return this;
  }

  /**
   * Get routeServiceStatus
   * @return routeServiceStatus
   **/
    public V3RouteServiceStatus getRouteServiceStatus() {
    return routeServiceStatus;
  }

  public void setRouteServiceStatus(V3RouteServiceStatus routeServiceStatus) {
    this.routeServiceStatus = routeServiceStatus;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3ResultRoute v3ResultRoute = (V3ResultRoute) o;
    return Objects.equals(this.routeName, v3ResultRoute.routeName) &&
        Objects.equals(this.routeNumber, v3ResultRoute.routeNumber) &&
        Objects.equals(this.routeType, v3ResultRoute.routeType) &&
        Objects.equals(this.routeId, v3ResultRoute.routeId) &&
        Objects.equals(this.routeGtfsId, v3ResultRoute.routeGtfsId) &&
        Objects.equals(this.routeServiceStatus, v3ResultRoute.routeServiceStatus);
  }

  @Override
  public int hashCode() {
    return Objects.hash(routeName, routeNumber, routeType, routeId, routeGtfsId, routeServiceStatus);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3ResultRoute {\n");
    
    sb.append("    routeName: ").append(toIndentedString(routeName)).append("\n");
    sb.append("    routeNumber: ").append(toIndentedString(routeNumber)).append("\n");
    sb.append("    routeType: ").append(toIndentedString(routeType)).append("\n");
    sb.append("    routeId: ").append(toIndentedString(routeId)).append("\n");
    sb.append("    routeGtfsId: ").append(toIndentedString(routeGtfsId)).append("\n");
    sb.append("    routeServiceStatus: ").append(toIndentedString(routeServiceStatus)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: dart/lib/model/v3_result_route.dart
part of swagger.api;

class V3ResultRoute {
  /* Name of route */
  String routeName = null;
/* Route number presented to public (nb. not route_id) */
  String routeNumber = null;
/* Transport mode identifier */
  int routeType = null;
/* Route identifier */
  int routeId = null;
/* GTFS Identifer of the route */
  String routeGtfsId = null;

  V3RouteServiceStatus routeServiceStatus = null;

  V3ResultRoute();

  @override
  String toString() {
    return 'V3ResultRoute[routeName=$routeName, routeNumber=$routeNumber, routeType=$routeType, routeId=$routeId, routeGtfsId=$routeGtfsId, routeServiceStatus=$routeServiceStatus, ]';
  }

  V3ResultRoute.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    routeName = json['route_name'];
    routeNumber = json['route_number'];
    routeType = json['route_type'];
    routeId = json['route_id'];
    routeGtfsId = json['route_gtfs_id'];
    routeServiceStatus = new V3RouteServiceStatus.fromJson(json['route_service_status']);
  }

  Map<String, dynamic> toJson() {
    return {
      'route_name': routeName,
      'route_number': routeNumber,
      'route_type': routeType,
      'route_id': routeId,
      'route_gtfs_id': routeGtfsId,
      'route_service_status': routeServiceStatus
     };
  }

  static List<V3ResultRoute> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3ResultRoute>() : json.map((value) => new V3ResultRoute.fromJson(value)).toList();
  }

  static Map<String, V3ResultRoute> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3ResultRoute>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3ResultRoute.fromJson(value));
    }
    return map;
  }
}
